File: javascript/hamming/javascript/hamming/hamming.js
exports.compute = function(dna1, dna2) {
    var words_a = dna1.split('');
    var words_b = dna2.split('');
    var distance = 0;
    if (words_a.length !== words_b.length) {
        throw new Error("DNA strands must be of equal length.");
    }
    for (word_index in words_a) {
        if (words_a[word_index] !== words_b[word_index]) {
            distance += 1;
        }
    }
    return distance;
};

File: lua/hamming/hamming.lua
Hamming = {}

function Hamming.compute(dna, dna2)
    --gmatch returns functions WTF lua?
    words_a = {};
    words_b = {};
    for i in string.gmatch(dna, ".") do
        words_a[#words_a + 1] = i;
    end
    for i in string.gmatch(dna2, ".") do
        words_b[#words_b + 1] = i;
    end  
    distance = 0
    for i, a in ipairs(words_a) do
        if a ~= words_b[i] and words_b[i] then
            distance = distance + 1
        end
    end
    return distance
end

return Hamming
